# Problem Set 1, Part Two: Due Tuesday, January 28 by 8am Eastern Standard Time

## Name: David Millard

**Show your work on all problems!** Be sure to give credit to any
collaborators, or outside sources used in solving the problems. Note
that if using an outside source to do a calculation, you should use it
as a reference for the method, and actually carry out the calculation
yourself; it’s not sufficient to quote the results of a calculation
contained in an outside source.

Fill in your solutions in the notebook below, inserting markdown and/or code cells as needed.  Try to do reasonably well with the typesetting, but don't feel compelled to replicate my formatting exactly.  **You do NOT need to make random variables blue!**

In [1]:
%matplotlib inline

In [2]:
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
plt.rcParams['figure.figsize'] = (8.0,5.0)
plt.rcParams['font.size'] = 14

### Bonus for Correct Filename

Your submitted version of the notebook should have a filename `ps01_2_lastname.ipynb` where `lastname` should be replaced by your last name, in all lowercase letters.  You'll get a bonus point here if this was done correctly.

### Empirical Confidence Interval Checking

**(a)** In this problem, we’ll be using the Central Limit Theorem to
    construct a confidence interval using the normal distribution, with
    endpoints ${{\overline{x}}}\pm z_{1-\alpha/2} s/\sqrt{n}$. We will
    carry out $N=10^4$ replications of an experiment which involves
    drawing a sample of size $n=50$ to construct a confidence interval
    of confidence level $1-\alpha=90\%$ on the mean $\mu$ of the
    sampling distribution. The fraction of those $N$ confidence
    intervals containing the true mean $\mu$ (which we will arbitrarily
    set to 5 for this experiment) should be about 90% if the procedure
    is correct. We can set things up with a few standard commands

In [3]:
N = 10**4
n = 50
mu = 5.
alpha = 1. - 0.90
zcrit = stats.norm.isf(0.5*alpha)

If each interval has a probability of 90% of containing the true
    value, what is the total number we expect out of $10,\!000$?

$10000 \cdot 0.9 = 9000$

What’s the standard deviation associated with that expectation?

$\sqrt{10000 \cdot 0.9 \cdot (1-0.9)} = 30$

**(b)** Let’s check this with a normal sampling distribution, which we can
    initialize with

In [4]:
mydist = stats.norm(loc=mu)

We can draw a sample (seeding the random number generator so the values are reproducible) with

In [5]:
np.random.seed(20190904)
x_Ii = mydist.rvs(size=(N,n))

and calculate the summary statistics with

In [6]:
xbar_I = x_Ii.mean(axis=-1)
s_I = x_Ii.std(axis=-1,ddof=1)

The `ddof` stands for “degenerate degrees of freedom”, which
    basically just means to use $n-1$ rather than $n$ in the
    normalization of the sample variance. We can define the ends of the
    confidence interval with

In [7]:
CIlo_I = xbar_I - zcrit*s_I/np.sqrt(n)
CIhi_I = xbar_I + zcrit*s_I/np.sqrt(n)

To check how many intervals contain the true value of $\mu=5$, we
    can use the construction that `CIlo_I<mu` will return an array of
    `True` and `False` values based on whether the inequality is true
    for each element of `CIlo_I`.

In [8]:
inCI_I = np.logical_and(CIlo_I<mu, mu<CIhi_I)

We can get the total number of such intervals with

In [9]:
print(np.sum(inCI_I))

8954


and the fraction with

In [10]:
print(np.mean(inCI_I))

0.8954


What fraction of the $N$ confidence intervals contain $\mu$?

$\frac{8954}{10000}$

Each of
    the $N$ confidence intervals has a different width, but display
    `CIhi_I-CIlo_I`, and you’ll see the first and last few.

In [11]:
CIhi_I-CIlo_I

array([0.44187625, 0.51064227, 0.44179558, ..., 0.39755524, 0.45816846,
       0.45220293])

What is the
    median with among the $N$ widths?

In [17]:
np.median(CIhi_I-CIlo_I)

0.4614319309578452

The median is approximately 0.4614.

**(c)** Repeat the process for a Laplace sampling distribution.

In [18]:
mydist = stats.laplace(loc=mu)

**(d)** Repeat the process for a Student-$t$ sampling distribution with $\nu=3$.
    (Hint: use `mydist = stats.t(loc=mu,df=3)`).

In [19]:
mydist = stats.t(loc=mu,df=3)

**(e)** Repeat the process for a Cauchy sampling Distribution.  What goes wrong in this case?

In [20]:
mydist = stats.cauchy(loc=mu)